receiver/varnishreceiver/client_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package varnishreceiver
+
+import (
+	"errors"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.uber.org/zap"
+)
+
+func TestNewVarnishClient(t *testing.T) {
+	client := newVarnishClient(
+		createDefaultConfig().(*Config),
+		componenttest.NewNopHost(),
+		componenttest.NewNopTelemetrySettings())
+	require.NotNil(t, client)
+}
+
+func TestBuildCommand(t *testing.T) {
+	testCases := []struct {
+		desc    string
+		config  Config
+		command string
+		argList []string
+	}{
+		{
+			desc: "without exec dir and with default host",
+			config: Config{
+				CacheDir: "defaultHostName",
+			},
+			command: "varnishstat",
+			argList: []string{"-j", "-n", "defaultHostName"},
+		},
+		{
+			desc: "without exec dir and with cache dir",
+			config: Config{
+				CacheDir: "/path/varnishinstance",
+			},
+			command: "varnishstat",
+			argList: []string{"-j", "-n", "/path/varnishinstance"},
+		},
+		{
+			desc: "with exec dir and cache dir",
+			config: Config{
+				CacheDir: "/path/varnishinstance",
+				ExecDir:  "/exec/dir/varnishstat",
```

Comment:
This isn't using a directory like the configuration says. It is testing using the full executable file path. See the other comment for the code change needed.

Reply:
removed varnishstat included in the configs ExecDir